desc/objectives/linear_objectives.py

```diff
@@ -179,7 +179,780 @@ class FixBoundaryZ(_Objective):
+        self._print_value_fmt = "Fixed-Z modes error: {:10.3e} (m)"
+
+    def build(self, eq, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        if self._modes is False or self._modes is None:  # no modes
+            modes = np.array([[]], dtype=int)
+            idx = np.array([], dtype=int)
+            modes_idx = idx
+            # FIXME: we don't want this option right? fix all modes...
+        elif self._modes is True:  # all modes
+            modes = eq.Z_basis.modes
+            idx = np.arange(eq.Z_basis.num_modes)
+            modes_idx = idx
+        else:  # specified modes
+            modes = np.atleast_2d(self._modes)
+            dtype = {
+                "names": ["f{}".format(i) for i in range(3)],
+                "formats": 3 * [modes.dtype],
+            }
+            _, idx, modes_idx = np.intersect1d(
+                eq.Z_basis.modes.astype(modes.dtype).view(dtype),
+                modes.view(dtype),
+                return_indices=True,
+            )
+            self._idx = idx
+            if idx.size < modes.shape[0]:
+                warnings.warn(
+                    colored(
+                        "Some of the given modes are not in the basis, "
+                        + "these modes will not be fixed.",
+                        "yellow",
+                    )
+                )
+
+        self._dim_f = modes_idx.size
+        # use given targets and weights if specified
+        if self.target.size == modes.shape[0] and None not in self.target:
+            self.target = self._target[modes_idx]
+        if self.weight.size == modes.shape[0] and self.weight != np.array(1):
+            self.weight = self._weight[modes_idx]
+
+        # use axis parameters as target if needed
+        if None in self.target or self.target.size != self.dim_f:
+            self.target = eq.Z_lmn[self._idx]
+
+        self._check_dimensions()
+        self._set_dimensions(eq)
+        self._set_derivatives(use_jit=use_jit)
+        self._built = True
+
+    def compute(self, Z_lmn, **kwargs):
+        """Compute Fixed mode Z errors.
+
+        Parameters
+        ----------
+        Z_lmn : ndarray
+            Spectral coefficients of Z(rho,theta,zeta) .
+
+        Returns
+        -------
+        f : ndarray
+            Fixed mode Z errors.
+
+        """
+        fixed_params = Z_lmn[self._idx]
+        return self._shift_scale(fixed_params)
+
+    @property
+    def target_arg(self):
+        """str: Name of argument corresponding to the target."""
+        return "Z_lmn"
+
+
+class FixSumModesR(_Objective):
```

Comment:
Would it be possible to combine this with the FixModesR class? We could just detect based on whether ``modes`` is a list or modes or a list of lists of modes

Reply:
Yes, I just wanted to make a simple way to start getting results this week